### PR TITLE
fix(auth): Cancel sign in

### PR DESCRIPTION
**Pull request by dnys1**
Properly enqueues sign-in calls and cancels previous attempts when signing in or confirming a sign in.

### CODE REVIEW

File: packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart [hunk 1]
```diff
@@ -552,59 +552,49 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
...
-            parameters: SignInParameters(
-              (p) => p
-                ..username = username
-                ..password = password,
-            ),
-            clientMetadata: pluginOptions.clientMetadata,
+    // Cancel the current sign in, if any.
+    await _stateMachine.acceptAndComplete<SignInNotStarted>(
+      const SignInEvent.cancelled(),
+    );
```
Comment (NikaHsn) [inline packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart:563]: `SignInStateMachine.resolve()` calls `_assertSignedOut()` and `_reset()` before running the `SignInEvent.initiate`. why do we need to cancel sign in flow if any, isn't it taken care by the SignInStateMachine?

File: packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart [hunk 2]
```diff
@@ -59,7 +59,7 @@ class SignInStateMachine extends AuthStateMachine<SignInEvent, SignInState> {
   }();
 
   /// Parameters to the flow.
-  late final SignInParameters parameters;
```
[the remark this answers is not in the document]
Comment (dnys1) [reply packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart:62]: The same state machine can be used for multiple sign ins.

File: packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart [hunk 3]
```diff
@@ -614,28 +604,22 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
               signInStep: AuthSignInStep.done,
             ),
           );
```
[the remark this answers is not in the document]
Comment (dnys1) [reply packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart]: Done 👍

File: packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart (hunk 1 above)
Comment (dnys1) [reply packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart]: Hmm yes this was leftover from earlier iterations. Removed 👍

**Review by Jordan-Nelson — APPROVED** [review]

**Review by NikaHsn — APPROVED** [review]

**Pull request merged**